### DISCUSSION TITLE
Do transforms work with lights?

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3846] [Q&A] [answered]
Are the position and direction of a spot light affected by the transform associated with the entity of the light?  Unless I've made a mistake, it looks like the answer is No, but that surprises me.

**Comment by romainguy** [comment, marked as answer]
You can transform lights. Some of our samples do it (see lightbulb.cpp for instance, it has a "disco ball" that rotates a bunch of spot lights).

**Reply by jwwalker** [reply]
OK, thanks.  One thing I was missing is that I wasn't using `TransformManager` to create a transform component for the light entity.  But I was also confused by the fact that applying a parent transform to the transform of a light does not affect what is returned by `LightManager::getPosition`, which says it returns a position in world space.  Seems like it should really be called local space, if it's the position before transforms have been accounted for.
